### ISSUE TITLE
Add option to turn off comments in homefeed

### ISSUE THREAD

**Issue by zcdunn** [forem/forem#8073]
**Is your feature request related to a problem? Please describe.**

I don't like seeing comments in the homefeed. I find them distracting, since seeing an out of context comment isn't useful. And when I'm in the homefeed, I scroll through to find posts to read. I'm not interested in comments yet.

**Describe the solution you'd like**

I'd like a setting to remove these comments.

**Describe alternatives you've considered**

**Additional context**

**Comment by rhymes** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by ludwiczakpawel** [comment]
@zcdunn thanks for this feedback. I think it's valid feedback. I personally sometimes get annoyed when there's lots of long comments... I'm not a huge fan of making this a user preference (although, i'm not saying _no_) but I wonder what would you say about truncating comments after 3 lines or so? So this post:

![image](https://user-images.githubusercontent.com/108287/83031043-64e9aa00-a034-11ea-90ac-06727ef48abb.png)

...would look like this:

![image](https://user-images.githubusercontent.com/108287/83031190-8fd3fe00-a034-11ea-90bd-45b408e4cdc9.png)

Or are you more in the "I don't want comments at all" team?

**Comment by benhalpern** [comment]
I'd prefer making this a user preference to default truncation, which would be really difficult to achieve effectively across all the types of content that can show up in a comment (like a picture or a liquid tag).

With user preferences I'm not sure I'd go as granular as "turn off/on comments" as I think it's more of a "card-only vs rich-feed, vs compact, etc." so the preference is a concept we can adjust the meaning to down the road.

Different communities would naturally have different default feeds I think, so giving options would satisfy that need while also allowing users to change themselves to use the same theme style across all communities.

**Comment by ludwiczakpawel** [comment]
@benhalpern yea, this kind of a toggle between different types of feed could be potentially interesting idea and agree - it could be very high level so we can always adjust it the way we want.. kind of like gmail does with display denesity: https://d.pr/v/nHoNED

Not sure where that goes into our timeline though :D

**Comment by benhalpern** [comment]
Yeah, the gmail example is perfect.

**Comment by zcdunn** [comment]
I'd definitely prefer a user preference to truncation. When I'm looking at my feed, I'm just looking at headlines and opening them in new tabs, so any comments are a distraction.

Thanks yall

**Comment by DRBragg** [comment]
I definitely agree with @zcdunn that the comments can be a distraction and I'd love the ability to turn the feature off. I think @benhalpern has the right idea about having multiple preference options for how the comments do or do not display. @ludwiczakpawel's example of the Gmail density setting is spot on.

This comment is basically a long winded way of me following this issue cause I'd love to work on it when it's available 😁

**Comment by deepu105** [comment]
copying over my comment from #8907 since I opened it without knowing about this issue

> I personally do not want to see top comments on posts on the feed as it makes eyeballing through the feed for interesting topics less pleasant. I can see that there will be people who like this, so it would be nice if I can turn off/on this view from settings or something to improve personal UX preference.

> It almost feels like Dev UX is slowly turning into something like Reddit :crying_cat_face:

Truncated comments are still distracting and honestly, I don't find comments helpful in any way personally. The previous UI of Dev was perfect IMO. Again since UX is always subjective, IMO its best to make it configurable where possible so individuals can set it the way they like. And yes the Gmail example is spot on

It could be a setting under the UX section of user settings, something like

- Feed post card view
   - **Simple**: Just post header, meta info and reactions
   - **Descriptive**: Simple + Post description or gist
   - **Comments**: Simple + top comments
   - **Detailed**: Descriptive + top comments

**Comment by ludwiczakpawel** [comment]
@deepu105 yea, we're pretty much leaning towards similar solution but I don't know yet what will be the possible options and how they gonna look like. I'm still not 100% sure where we gonna implement that, but we will definitely do it :)

**Comment by cmgorton** [comment]
Has this issue been fixed @ludwiczakpawel ? I just realized today I don't see comments on all the posts anymore. Wondering if this issue can be closed?

**Comment by ludwiczakpawel** [comment]
@cmgorton No, I don't think it has been fixed.

The reason why you don't see comments in the feed is because we might have adjusted criteria for when comments show in the feed (min score required for that) - but that's just my guess. Another option is we have less high-score comments right now. And another option - something is broken :d  (cc @joshpuetz may know something?)

If you go to "Week / Month / Year / Infinity" you will certainly see some comments there.

In other words, this feature request has not been implemented, and definitely not in a way that was proposed in the comments.

**Comment by cmgorton** [comment]
> @cmgorton No, I don't think it has been fixed.
> 
> The reason why you don't see comments in the feed is because we might have adjusted criteria for when comments show in the feed (min score required for that) - but that's just my guess. Another option is we have less high-score comments right now. And another option - something is broken :d (cc @joshpuetz may know something?)
> 
> If you go to "Week / Month / Year / Infinity" you will certainly see some comments there.
> 
> In other words, this feature request has not been implemented, and definitely not in a way that was proposed in the comments.

Ok I will leave it open then. Thank you for the explanation :)

**Comment by cmgorton** [comment]
Thank you for opening this request.
Features tend to be subjective and might spur some debate. This one has been around a while with not much traction.
The Forem core team now uses an internal RFC ("request for comments") process to assess and prioritize new features. This process is intended to provide a consistent and standardized path for new changes to enter the Forem ecosystem. If you'd like to propose a new feature like this one, please visit [forem.dev](https://forem.dev/) to start a discussion around it! 🙏
